test/extended/router/certs.go

```diff
@@ -76,40 +76,45 @@ J5dJXQxp7y20vh2C1e3wTPlA5CHHKpU1JZAe4THCJUg+EPwa4I+BOlvp71EB7BaH
 bk65iLoLrUSkxMDi46qTAs5K
 -----END PRIVATE KEY-----
 `
+	// Commands to recreate this cert:
+	//    openssl genrsa -out testkey.pem 1024
+	//    openssl req -new -x509 -sha256 -key testkey.pem -out testcert.pem -days 3650 -subj "/C=US/ST=NC/L=Default City/O=Default Company Ltd/OU=Test CA/CN=www.exampleca.com/emailAddress=example@example.com"
+	//    cat testcert.pem testkey.pem
```

Comment:
This results in a certificate with `CA:TRUE`:
```text
% openssl x509 -noout -text -in testcert.pem                                                                   
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            4d:46:b2:d8:e6:36:b0:3d:70:84:8d:c0:d9:8d:a2:0f:ce:4d:83:45
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C = US, ST = NC, L = Default City, O = Default Company Ltd, OU = Test CA, CN = www.exampleca.com, emailAddress = example@example.com
        Validity
            Not Before: Dec 13 21:47:06 2023 GMT
            Not After : Dec 10 21:47:06 2033 GMT
        Subject: C = US, ST = NC, L = Default City, O = Default Company Ltd, OU = Test CA, CN = www.exampleca.com, emailAddress = example@example.com
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (1024 bit)
                Modulus:
                    00:9b:5d:54:ab:76:23:46:bf:4c:ef:49:ce:d3:b3:
                    d5:34:b4:97:ac:e7:25:6a:7e:74:53:7a:82:24:67:
                    6e:89:00:f1:77:62:25:1f:71:95:ae:f6:42:60:4d:
                    5f:34:01:19:86:d3:2d:eb:9e:61:83:53:01:a2:0b:
                    15:cd:c8:29:66:4a:3f:cf:99:5f:a7:25:0b:a7:72:
                    7e:60:60:21:81:3b:fe:e5:ab:64:8d:b5:fd:c2:0b:
                    25:ea:43:e3:bf:3b:83:ac:15:30:c4:ff:ca:f5:72:
                    3e:4d:70:a2:a6:4c:79:81:73:b9:57:96:15:b9:c8:
                    d3:28:25:b5:e4:ac:f1:e4:2d
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Subject Key Identifier: 
                FF:99:34:65:BD:56:9C:7D:D5:92:06:A6:69:FC:19:9A:AE:1E:69:4D
            X509v3 Authority Key Identifier: 
                keyid:FF:99:34:65:BD:56:9C:7D:D5:92:06:A6:69:FC:19:9A:AE:1E:69:4D

            X509v3 Basic Constraints: critical
                CA:TRUE
    Signature Algorithm: sha256WithRSAEncryption
         76:d5:a5:f2:f0:e4:1e:2d:66:26:78:53:cb:e8:61:e1:af:ab:
         3b:3a:0f:58:f0:66:fc:7f:dd:5f:90:be:cd:f5:d1:dc:88:02:
         ad:08:25:c8:cf:62:1a:a5:41:3c:65:fe:90:80:d8:d1:ec:53:
         e6:23:e7:17:67:83:66:20:42:64:5a:02:79:3a:6a:48:f5:2e:
         b3:54:a8:cf:fb:ee:91:b4:b2:f4:32:46:58:d9:a3:e3:f5:8d:
         9c:fa:65:a9:27:26:38:41:ef:71:a0:68:91:5e:30:4a:50:35:
         5e:c1:e6:95:72:01:48:68:4a:11:35:5d:70:0f:0a:9f:2a:17:
         7b:ad
% 
```
Setting the X509v3 extensions when using `openssl req` is tricky, but I think the following commands create a proper certificate:
```bash
openssl req -x509 -newkey rsa:1024 -days 3650 -keyout exampleca.key -out exampleca.crt -nodes -subj '/C=US/ST=SC/L=Default City/O=Default Company Ltd/OU=Test CA/CN=www.exampleca.com/emailAddress=example@example.com'                                                                                                                                                                                                               
openssl req -newkey rsa:1024 -nodes -keyout example.key -out example.csr -subj '/CN=www.example.com/ST=SC/C=US/emailAddress=example@example.com/O=Example/OU=Example'
openssl x509 -req -days 3650 -in example.csr -signkey example.key -CA exampleca.crt -CAcreateserial -CAkey exampleca.key -clrext -extensions ext -extfile <(echo $'[ext]\nbasicConstraints = CA:FALSE') -out example.crt
```
This gives me the following certificate, which matches the old one except for the serial number, signature algorithm (which is what we want to change), validity period, and key and signature:
```text
% openssl x509 -noout -text -in example.crt 
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            3c:51:e8:de:4a:db:34:32:77:e3:61:ce:35:b9:e0:3f:b7:9d:92:31
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C = US, ST = SC, L = Default City, O = Default Company Ltd, OU = Test CA, CN = www.exampleca.com, emailAddress = example@example.com
        Validity
            Not Before: Dec 18 22:22:20 2023 GMT
            Not After : Dec 15 22:22:20 2033 GMT
        Subject: CN = www.example.com, ST = SC, C = US, emailAddress = example@example.com, O = Example, OU = Example
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (1024 bit)
                Modulus:
                    00:b8:bf:4b:a9:e9:9b:49:4a:bb:fd:a6:b6:75:15:
                    30:71:00:98:bb:ee:41:f5:72:eb:ac:89:9f:5d:7c:
                    fd:d8:d4:e5:9a:da:c8:34:a0:43:d7:da:4f:5b:34:
                    79:b1:cd:9a:32:13:e0:9e:bd:fb:96:00:da:4e:93:
                    9a:3e:2e:4e:5f:f6:60:b5:ac:3d:b1:c4:c0:17:3d:
                    ce:63:e3:4a:51:8e:23:4b:70:3f:8e:6c:8d:f9:16:
                    a7:4e:73:c6:96:0a:fc:63:06:37:34:8c:59:cb:14:
                    20:08:a2:7f:7a:5b:83:e2:f6:77:0a:cf:d9:40:de:
                    a0:e6:79:5a:bf:9b:94:d0:61
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Basic Constraints: 
                CA:FALSE
    Signature Algorithm: sha256WithRSAEncryption
         ca:fe:38:66:c8:4a:51:9a:be:4b:43:0d:90:f0:0e:cb:31:a2:
         55:53:82:af:d7:78:7d:97:0b:34:03:56:b6:fc:d2:35:83:b9:
         dc:6d:a5:0e:22:25:81:51:24:2a:cd:90:5e:4f:45:0a:16:5f:
         a2:40:21:1a:1c:f7:81:0e:5d:22:21:5c:c7:49:d9:52:4a:a4:
         b0:6d:2a:5d:f6:9e:a2:29:01:96:10:fd:46:8f:50:59:7e:43:
         6e:59:2d:25:ab:a1:00:bf:7c:80:47:6e:a0:03:8f:a6:25:0e:
         74:e9:15:08:7c:94:54:fe:19:fc:98:44:62:c6:9f:fb:82:de:
         96:11
% 
```